helper/resource/plan_check.go

```diff
@@ -0,0 +1,44 @@
+package resource
+
+import (
+	"github.com/hashicorp/go-multierror"
+	tfjson "github.com/hashicorp/terraform-json"
+	"github.com/mitchellh/go-testing-interface"
+)
+
+// TODO: document
+type PlanCheck interface {
+	RunCheck(req PlanCheckRequest, resp *PlanCheckResponse)
```

Comment:
Recommend adding `context.Context` to prevent a later breaking change with https://github.com/hashicorp/terraform-plugin-testing/issues/88 (we do not need/want to solve that issue as part of this enhancement)

```go
	RunCheck(context.Context, PlanCheckRequest, *PlanCheckResponse)
```

Although we may want to further consider having "Plan" in the name for a future where there could be other checks such as "better" state checks. e.g.

```go
	CheckPlan(context.Context, CheckPlanRequest, *CheckPlanResponse)
```